### PR TITLE
feat: add Infinite Scroll to home page

### LINKED ISSUE #1091: [FEATURE] Infinite scroll
### Description

the first page loads all the users at a time, would be cool to add Infinite Scroll that automatically adds the next page when hiting the bottom, saving users from a full page load if there is plenty of users to diplay.

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Hugher256**
## Fixes Issue

Closes #1091

## Changes proposed

Revamp of the filtering logic to accommodate for the infinite scrolling functionality. Not fetching API for all the profiles rather only the profiles shown on screen.

## Check List (Check all the applicable boxes)

- [X] My code follows the code style of this project.
- [] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [X] The title of my pull request is a short description of the requested changes.

## Note to reviewers

Some of the test cases were not passing as this PR revamps the rendering logic. 'search.feature' test was not able to find profile for 'Eddie Jaoude' as the profile was not rendering due to limited initial rendering due to infinite scroll. Similarly 'user.feature' was not able to find the profile due to same reason.

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/1201"></a>

### CODE REVIEW

**Review by vikasganiga05 — CHANGES_REQUESTED** [review]
Please fix those failing tests.

**Comment by schmelto** [comment]
> Please fix those failing tests.

```
Running:  search.feature                                                                  (3 of 4)
130
[0] 
131
[0] 
132
[0]   Search
133
[0]     ✓ Search on the homepage (3253ms)
134
[0]     1) Search with results
135
[0]     2) Search with no results
136
[0] 
137
[0] 
138
[0]   1 passing (14s)
139
[0]   2 failing
140
[0] 
141
[0]   1) Search
142
[0]        Search with results:
143
[0]      AssertionError: Timed out retrying after 4000ms: expected '<main>' to contain 'Eddie Jaoude'

```

**Comment by schmelto** [comment]
## GitHub Actions are failing

### TLDR;

Thank you for the PR. There is a GitHub Action failing, please check the Action output to find out more about the error.

---

### 

GitHub Actions perform automated tests. They do not replace manual review but help us focus on the more valuable collaboration.

GitHub Actions usually include...
- Building the project
- Automated testing
- Linting to check consistent in code formatting

---

- Languages: all
- Labels: Style guide

__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix

**Comment by Hugher256** [comment]
I will make the necessary changes.

**Comment by Hugher256** [comment]
@schmelto Can I please request a manual review? I have looked into the changes and as I have already explained the test that is failing is search.feature as it is unable to see 'Eddie Jaoude'. This is due to the element not initially rendering as I have limited initial rendering to implement infinite scroll.

**Comment by Hugher256** [comment]
I have updated the tests to incorporate infinite scrolling. But it is not letting me commit as in actions.js `cy` is not invoked and husky is failing my commit. Any help in this case on how I can commit actions.js?

**Review by schmelto — APPROVED** [review]
## Looking good to me

Your suggested changes looking good to me.
__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :ok_hand: This code looks good&nbsp; | &nbsp;**Tags:** Elegant

**Review by vikasganiga05 — APPROVED** [review]
LGTM :+1: Thanks for your contribution :tada:

**Review by krshdev — APPROVED** [review]

**Comment by emmalearnscode** [comment]
Thanks for your contribution. We are currently discussing if this is something we want to add or whether we are going to revamp the landing page in a different way without the user chips which is why this hasn't been merged yet.

__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :question: I have a question

**Comment by Hugher256** [comment]
Thank you for letting me know. I will look for other issue to work on.

**Comment by krshkodes** [comment]
Hi, @emmalearnscode if I am not wrong hasn't the other landing been chosen and built as well.

__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :question: I have a question

**Comment by emmalearnscode** [comment]
Thanks so much for contributing this pull request. The code looks good, however, we've decided to take the project in a different direction and solved the issue in a different way and therefore this change is no longer needed.

Please feel free to help out with other issues in the future.
